File: QytetetRuby/lib/dado.rb
#encoding: utf-8
require "singleton"

module ModeloQytetet
  class Dado
    include Singleton 
    
    attr_reader :valor
    
    def initialize
      
    end
    
    def tirar
      numero = Random.new
      
      @valor = 1 +numero.rand(6)
    end
    
  end
end

File: QytetetJava/src/modeloqytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.Random;

/**
 *
 * @author Juan Carlos Hermoso Quesada
 */
public class Dado {
    private int valor;
    private static final Dado instance = new Dado();
    
    private Dado() {
        
    }
    
    
    public static Dado getInstance() {
        return instance;
    }
    
    int tirar() {
        int valor_dado = (int)(Math.random() * 6) + 1;
        
        this.valor = valor_dado;
        
        return valor_dado;
    }
    
    public int getValor() {
        return valor;
    }
}
